src/modules/ekf2/EKF/output_predictor/output_predictor.cpp

```diff
@@ -244,9 +246,14 @@ void OutputPredictor::calculateOutputStates(const uint64_t time_us, const Vector
 	}
 
 	// update auxiliary yaw estimate
-	const Vector3f unbiased_delta_angle = delta_angle - delta_angle_bias_scaled;
-	const float spin_del_ang_D = unbiased_delta_angle.dot(Vector3f(_R_to_earth_now.row(2)));
-	_unaided_yaw = matrix::wrap_pi(_unaided_yaw + spin_del_ang_D);
+	// rotate the state quternion by the delta quaternion only corrected for bias without EKF corrections
+	const Vector3f delta_angle_unaided = delta_angle - delta_angle_bias_scaled;
+	const float yaw_state = Eulerf(quat_nominal_before_update).psi();
+	const Quatf quat_unaided = quat_nominal_before_update * Quatf(AxisAnglef(delta_angle_unaided));
+	const float yaw_without_aiding = Eulerf(quat_unaided).psi();
+	// Yaw before delta quaternion applied and yaw after. The difference is the delta yaw. Accumulate it.
+	const float yunaided_yaw_update = yaw_without_aiding - yaw_state;
+	_unaided_yaw = matrix::wrap_pi(_unaided_yaw + yunaided_yaw_update);
```

Comment:
```suggestion
	_unaided_yaw = matrix::wrap_pi(_unaided_yaw + unaided_delta_yaw);
```

Reply:
Same as the comment above https://github.com/PX4/PX4-Autopilot/compare/ce727ea97ab88c5302e46a5c4746b021333eed3f..07f5048cb9f4725177038aba4321243505d92f72#diff-2ee4cc51742f7487431716ca452262215e942f98c029f02954af5b6cdbbc1e01R255-R256